### DISCUSSION TITLE
MFA support?

### DISCUSSION THREAD

**Discussion by BrianVallelunga** [datalust/seq-tickets#1717] [Help] [answered]
I don't currently see any way to add multi-factor authentication to Seq's users. Is this planned?

**Comment by nblumhardt** [comment, marked as answer]
Hi Brian! Thanks for reaching out.

In general we're expecting organizations that want to use 2FA to also be using SSO, and thus be relying on Azure AD or an OIDC provider for this. Is that an option for you?

Best regards,
Nick

**Comment by BrianVallelunga** [comment]
Thanks for the reply. That makes sense and I just realized we could integrate it with our Auth0 we already have. Out of curiosity, does OIDC support Google Workspaces/Apps?

**Reply by nblumhardt** [reply]
That's great 👍 

RE Google auth, I believe it does (we definitely have customers using OIDC with Google auth), although I don't know all of the ins-and-outs of Google's offering.

**Comment by CumpsD** [comment]
Any plans on supporting regular 2FA? Things like Authy.

ChatGPT already thinks you have it ;)

![image](https://user-images.githubusercontent.com/1201644/228842992-79166b96-0d22-49d5-8df9-23c4acf06085.png)

**Reply by nblumhardt** [reply]
LOL - thanks ChatGPT! :-)

It'd be good to know what kind of environment/situation this would be most useful in  - e.g. solo dev? Or some disconnected scenarios? So far, in the vast majority of cases, 2FA and SSO have seemed to go hand-in-hand, so building a better picture here would help us figure out if it's something we can schedule in.

**Reply by CumpsD** [reply]
This would be Seq hosted on a k8s cluster with public UI access, but without a SSO solution in place.

Right now every user has a user/pass combo, but we try to secure as much as we can with 2FA. Mostly TOTP (Authy), sometimes also hardware key (Yubikey).
